servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscovererEvent.java

```diff
@@ -26,10 +32,105 @@
+        throw new UnsupportedOperationException("Migrate to status() method. This method may be implemented" +
+                " temporarily until migration to status() is complete.");
+    }
+
+    /**
+     * Status provided by the {@link ServiceDiscoverer} system that guides the actions of {@link LoadBalancer} upon the
+     * bound {@link ServiceDiscovererEvent#address()} (via {@link ServiceDiscovererEvent}).
+     */
+    final class Status {
+
+        /**
+         * Signifies the {@link ServiceDiscovererEvent#address()} is available for use in connection establishment.
+         */
+        public static final Status AVAILABLE = new Status("available");
+
+        /**
+         * Signifies the {@link ServiceDiscovererEvent#address()} is not available for use and all currently established
+         * connections should be closed.
+         */
+        public static final Status UNAVAILABLE = new Status("unavailable");
+
+        /**
+         * Signifies the {@link ServiceDiscovererEvent#address()} is expired and should not be used for establishing
+         * new connections. It doesn't necessarily mean that the host should not be used in traffic routing over already
+         * established connections as long as they are kept open by the remote peer. The implementations can have
+         * different policies in that regard.
+         */
+        public static final Status EXPIRED = new Status("expired");
+
+        private final String name;
+
+        private Status(final String name) {
+            if (name.isEmpty()) {
+                throw new IllegalArgumentException("Status name cannot be empty");
+            }
+            this.name = name.toLowerCase(Locale.ENGLISH);
+        }
+
+        /**
+         * Returns an {@link Status} for the specified name.
+         * @param name the status name.
+         * @return {@link Status} representing the status for given name.
+         */
+        public static Status of(final String name) {
+            switch (name.toLowerCase(Locale.ENGLISH)) {
+                case "available":
+                    return AVAILABLE;
+                case "unavailable":
+                    return UNAVAILABLE;
+                case "expired":
+                    return EXPIRED;
+                default:
+                    return new Status(name);
+            }
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof ServiceDiscovererEvent.Status)) {
+                return false;
+            }
+            final Status that = (Status) o;
+            return name.equals(that.name);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(name);
+        }
+
+        @Override
+        public String toString() {
+            return "Status{" +
+                    "name='" + name + '\'' +
```

Comment:
True, but when a reference to a status is held not in context of an event, losing that contextual wrapper could confuse users when debugging and investigating, I prefer to keep it.

Reply:
Instead of your suggestion, I will go with adding a `public String getName()` method to `Status`. I believe with that `ServiceDiscovererEvent` can just use `Status#getName` in `toString` while preserving the desired properties described above.

Reply:
There is a high change that a `Status` instance will be passed to a logger. It's known that status has no other fields except `name` => not necessary to add `name=` part. Name of the class is also less useful. Debugger always shows the type, if users log they usually know what they log and do it with some other context info, ST core does not print it without an event. 
For the constants like this one, all other final classes just print it's main meaning. Have a look at `HttpRequestMethod`, `HttpResponseStatus`, `HttpProtocolVersion`.

Reply:
Ok, I am convinced with the "enum" part ;-)